drivers/SmartThings/philips-hue/src/utils.lua

```diff
@@ -64,10 +64,102 @@ function utils.backoff_builder(max, inc, rand)
   end
 end
 
+function utils.labeled_socket_builder(label)
+  local log = require "log"
+  local socket = require "cosock.socket"
+  local ssl = require "cosock.ssl"
+
+  label = (label or "")
+  if #label > 0 then
+    label = label .. " "
+  end
+
+  local function make_socket(host, port, wrap_ssl)
+    log.info(
+      string.format(
+        "%sCreating TCP socket for Hue REST Connection", label
+      )
+    )
+    local _ = nil
+    local sock, err = socket.tcp()
+
+    if err ~= nil or (not sock) then
+      return nil, (err or "unknown error creating TCP socket")
+    end
+
+    log.info(
+      string.format(
+        "%sSetting TCP socket timeout for Hue REST Connection", label
+      )
+    )
+    _, err = sock:settimeout(60)
+    if err ~= nil then
+      return nil, "settimeout error: " .. err
+    end
+
+    log.info(
+      string.format(
+        "%sConnecting TCP socket for Hue REST Connection", label
+      )
+    )
+    _, err = sock:connect(host, port)
+    if err ~= nil then
+      return nil, "Connect error: " .. err
+    end
+
+    log.info(
+      string.format(
+        "%sSet Keepalive for TCP socket for Hue REST Connection", label
+      )
+    )
+    _, err = sock:setoption("keepalive", true)
+    if err ~= nil then
+      return nil, "Setoption error: " .. err
+    end
+
+    if wrap_ssl then
+      log.info(
+        string.format(
+          "%sCreating SSL wrapper for for Hue REST Connection", label
+        )
+      )
+      sock, err =
+        ssl.wrap(sock, {mode = "client", protocol = "any", verify = "none", options = "all"})
+      if err ~= nil then
+         return nil, "SSL wrap error: " .. err
+      end
+      log.info(
+        string.format(
+          "%sPerforming SSL handshake for for Hue REST Connection", label
+        )
+      )
+        _, err = sock:dohandshake()
+      if err ~= nil then
+        return nil, "Error with SSL handshake: " .. err
+      end
+    end
+
+    log.info(
+      string.format(
+        "%sSuccessfully created TCP connection for Hue", label
+      )
+    )
+    return sock, err
+  end
+  return make_socket
+end
+
+-- This version of log_func_wrapper has slightly different
+-- characteristics and formatting than the one in the st/utils
+-- files.
+--
+-- Of note, this log_func_wrapper *also* sets hub_logs = true,
+-- so it should be used in chatty paths sparingly if using
+-- info level.
 function utils.log_func_wrapper(func, func_name, log_level)
```

Comment:
is this still used in the driver at all?

Reply:
It is not. I figured it could be useful to preserve in case we want to start using it again.

Reply:
Eh, I'd say gut it since the code exists in the main utils we can always pull it back in with the minor changes we use here.